### PR TITLE
bump openapigen commit and mocha to get native javascript url wrappers

### PR DESCRIPTION
**Pull request by davidgamero**
get our URL parsing to use the native URL instead of a deprecated dependency we inherited from OpenAPIGen
https://github.com/OpenAPITools/openapi-generator/pull/14319

also remove the last of our shims 'models/IntOrString' residual file that isn't needed since the import mapping works now!

### CODE REVIEW

**Comment by davidgamero** [comment]
looks like theres'a tslint v prettier issue atm, i'm investigating

**Comment by brendandburns** [comment]
@davidgamero this change lgtm once you sort out the linting.

**Comment by davidgamero** [comment]
ended up having to make the mocha jump from v5 to v10 to get the critical CVEs out, but it was just some config changes and some small interface tightening :)

**Comment by davidgamero** [comment]
ah yes, codeql thinks i both fixes and reintroduced the same certificate validation vuln in `config_test.ts`

**Comment by davidgamero** [comment]
@brendandburns idk how to exclude the test file/mark that this isn't a new issue for CodeQL, but we are good to merge now- linting and testing is all passing

File: src/config_test.ts [hunk 1]
```diff
@@ -265,24 +265,24 @@ describe('KubeConfig', () => {
...
             await kc.applytoHTTPSOptions(opts);
-            const expectedCA = Buffer.from('CADATA2', 'utf-8')
+            const expectedCA = Buffer.from('CADATA2', 'utf-8');
             const expectedAgent = new https.Agent({
                 ca: expectedCA,
                 cert: undefined,
                 key: undefined,
                 passphrase: undefined,
                 pfx: undefined,
                 rejectUnauthorized: false,
```
Comment (davidgamero) [inline src/config_test.ts:275]: this isn't an actual new vulnerability, codeql is triggering on a test that simulates password auth which has `rejectUnauthorized: false`

**Comment by davidgamero** [comment]
i believe this is good to go now @brendandburns

**Pull request closed without merging**